cli/src/main/java/com/devonfw/tools/ide/variable/IdeVariables.java

```diff
@@ -49,6 +49,18 @@ public interface IdeVariables {
 
   /** {@link VariableDefinition} for {@link com.devonfw.tools.ide.context.IdeContext#getWorkspaceName() WORKSPACE}. */
 
+  /** {@link VariableDefinition} for default build options of mvn */
+  VariableDefinitionString MVN_BUILD_OPTS = new VariableDefinitionString("MVN_BUILD_OPTS", null, c -> "");
+
+  /** {@link VariableDefinition} for default build options of npm */
+  VariableDefinitionString NPM_BUILD_OPTS = new VariableDefinitionString("NPM_BUILD_OPTS", null, c -> "");
```

Comment:
This one seems more tricky - see
https://github.com/devonfw/ide/blob/cdf4af5de21489f3fdcbfdab41af858822e0d598/scripts/src/main/resources/scripts/command/npm#L118-L124
and
https://github.com/devonfw/ide/blob/cdf4af5de21489f3fdcbfdab41af858822e0d598/scripts/src/main/resources/scripts/command/npm#L92-L98

Maybe we should address this one in a separate PR and merge as is? So far this would be my suggestion to make better progress... So either create an issue for this and link here as comment so I can resolve or already address in this PR adding even more new features and complexity.

Reply:
I've added a new issue and TODOs to these sections.